docs/src/content/docs/commands/JSON.STRLEN.md

```diff
@@ -1,43 +1,49 @@
+|-----------|-------------------------------------------------------------------------|--------|----------|
+| `key`     | The key under which the JSON document is stored.                        | String | Yes      |
+| `path`    | The JSONPath to the string within the JSON document. The path must contain a string.  | String | No       |
 
 ## Return Value
 
-- `Integer`: The length of the JSON string at the specified path.
-- `Null`: If the path does not exist or does not point to a JSON string.
+| Condition                                 | Return Value                                                                    |
+|-------------------------------------------|---------------------------------------------------------------------------------|
+| JSON string is found at the specified path| Length of the JSON string.                                                      |
+| JSONPath contains `*` wildcard            | Indicates the length of each key in the JSON. Returns `nil` for non-string keys.|
+
 
 ## Behaviour
 
 When the `JSON.STRLEN` command is executed, DiceDB will:
 
-1. Retrieve the JSON document stored at the specified key.
-2. Navigate to the specified path within the JSON document.
-3. If the path points to a JSON string, return the length of the string.
-4. If the path does not exist or does not point to a JSON string, return `null`.
+1. If the specified `key` does not exist, the command returns `null`.
+2. If `path` is not specified, the command throws an error.
+3. `$` is considered as root path which returns `null`.
+4. If the specified path exists and points to a JSON string, the command returns the length of the string.
+5. If the path does not exist or does not point to a JSON string, an error is returned.
+6. Multiple results are represented as a list in case of wildcards, where each string length is returned in order of appearance and `null` is returned if it's not a string.
 
-## Error Handling
+## Errors
 
 The following errors may be raised when executing the `JSON.STRLEN` command:
 
-- `(error) ERR wrong number of arguments for 'JSON.STRLEN' command`: This error occurs if the number of arguments provided to the command is incorrect.
-- `(error) ERR key does not exist`: This error occurs if the specified key does not exist in the DiceDB database.
-- `(error) ERR path does not exist`: This error occurs if the specified path does not exist within the JSON document.
-- `(error) ERR path is not a string`: This error occurs if the specified path does not point to a JSON string.
+- `(error) ERROR wrong number of arguments for 'json.strlen' command`: This error occurs if the number of arguments provided to the command is incorrect.
+- `(error)  ERROR invalid JSONPath`: This error occurs if the specified path does not exist within the JSON document.
+- `(error)  ERROR WRONGTYPE wrong type of path value - expected string but found integer`: This error occurs if valid key is provided but path is not provided.
```

Comment:
Hi, this error occurs if the value at a valid json path is an integer and not string. Please confirm

Reply:
![invalidPath](https://github.com/user-attachments/assets/8990dff0-aa46-495b-ac58-a253a5ed22e0)

Reply:
![Uploading noPath.jpg…]()

Reply:
![noPath](https://github.com/user-attachments/assets/8ac7d43b-b846-4dc4-ab4d-256069e5dc9f)

Reply:
Hi, I think this error occurs because the path, if not provided, is by default set to $ and if you set the value to an integer, it throws this error. Please change the decription

Reply:
Hey, I don't think that if path is not provided then it is taking '$' as default value because for any JSON object if I am not providing path it is throwing the above error and if '$' is provided then nil is returned as output

Reply:
gotcha, can you please share the scenarios you are running? I will check and see if we need to fix something. 
Thanks.

Reply:
gotcha, can you please share the scenarios you are running? I will check and see if we need to fix something.
Thanks.

Reply:
![Uploading test.jpg…]()

Reply:
![test](https://github.com/user-attachments/assets/b5d7848d-95e4-47f5-8963-49de3d06e8a4)

Reply:
Thanks

Reply:
I just checked and you are right about the behaviour. Basically it will take the object at root as string else throw error. There is an issue with the error message though. It is inconsistent with redis.
Redis:
<img width="619" alt="image" src="https://github.com/user-attachments/assets/707527eb-8a36-4f4a-ab77-fa0d5d2b7871">
DiceDB:
<img width="567" alt="image" src="https://github.com/user-attachments/assets/a8cdb34b-1718-4d29-bb67-589a94ea23c5">

I will create an issue for this. Let me know, if you would like to take this up as well since this issue is dependent on that one.

Thanks

Reply:
Hi @bhima2001 , Please comment on this https://github.com/DiceDB/dice/issues/1060 if you would like to work on this.
Thanks

Reply:
Hey @apoorvyadav1111 I have added  a comment to that PR

Reply:
Hi, I cannot assign the issue to you. But one of the maintainers will be doing the same soon. I would suggest closing that issue before this one or both of them together.